### PR TITLE
CV2-3397: add more than in date range filter

### PR DESCRIPTION
**Pull request by melsawy**
## Description

Add more than option to date range filter so user can filter by `less than` or `more than` options

References: CV2-3397

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

1. Go to list page.
2. Choose date range filter
3. Choose `More than` option

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I've made sure my branch is runnable and given good testing steps in the PR description
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] I have removed the /* eslint-disable @calm/react-intl/missing-attribute */ from any files I've worked on and added a `description` prop to all `<FormattedMessage />` components that were missing it.
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Review by caiosba — COMMENTED** [review]
@melsawy , I would like to try the code locally, can I use the `develop` branch from Check API or I need to use a specific branch from Check API that is not merged yet?

**Comment by melsawy** [comment]
@caiosba you should checkout this branch `CV2-3397-add-option-more-than-in-date-range`

File: src/app/components/search/DateRangeFilter.js [hunk 1]
```diff
@@ -305,9 +306,10 @@ const DateRangeFilter = ({
         getStartDateStringOrNull(),
         getEndDateStringOrNull(),
       ));
-    } else if (rangeType === rangeTypes.relative) {
+    } else if (['less_than', 'more_than'].indexOf(rangeType) !== -1) {
```
Comment (amoedoamorim) [inline src/app/components/search/DateRangeFilter.js:309]: I suggest changing this comparison to `['less_than', 'more_than'].includes(rangeType)` for better readability

File: src/app/components/search/DateRangeFilter.js [hunk 2]
```diff
@@ -324,9 +326,10 @@ const DateRangeFilter = ({
         getStartDateStringOrNull(),
         getEndDateStringOrNull(),
       ));
-    } else if (valueRangeType === rangeTypes.relative) {
+    } else if (['less_than', 'more_than'].indexOf(valueRangeType) !== -1) {
```
Comment (amoedoamorim) [inline src/app/components/search/DateRangeFilter.js:329]: I suggest changing this comparison to `['less_than', 'more_than'].includes(valueRangeType)` for better readability

File: src/app/components/search/SearchResults.js [hunk 3]
```diff
@@ -230,10 +230,11 @@ function SearchResultsComponent({
         oldQuery.range.updated_at ||
         oldQuery.range.media_published_at ||
         oldQuery.range.report_published_at || {};
-      if ((!datesObj.start_time && !datesObj.end_time) && (datesObj.condition !== 'less_than')) {
+      const relativeCondition = ['less_than', 'more_than'].indexOf(datesObj.condition);
```
Comment (amoedoamorim) [inline src/app/components/search/SearchResults.js:233]: I suggest changing this comparison to `['less_than', 'more_than'].includes(datesObj.condition)` and adjusting the logic accordingly, for better readability

**Review by amoedoamorim — APPROVED** [review]
Added a few suggestions for better readability, otherwise LGTM. Tried filter locally, working as expected!

**Review by caiosba — APPROVED** [review]
I tested it locally and works as expected.

File: src/app/components/search/DateRangeFilter.js (hunk 1 above)
Comment (melsawy) [reply src/app/components/search/DateRangeFilter.js]: Done

File: src/app/components/search/DateRangeFilter.js (hunk 2 above)
Comment (melsawy) [reply src/app/components/search/DateRangeFilter.js]: Done

File: src/app/components/search/SearchResults.js (hunk 3 above)
Comment (melsawy) [reply src/app/components/search/SearchResults.js]: Done

**Pull request merged**